python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -91,10 +92,27 @@ def struct_to_string(name: str, **kwargs: object) -> str:
     return f"{name}({props_str})"
 
 
-def validate_tags(
-    tags: Optional[Mapping[str, Any]], allow_reserved_tags: bool = True
-) -> Mapping[str, str]:
+class NormalizedTags(NamedTuple):
+    tags: Mapping[str, str]
+
+    def with_normalized_tags(self, normalized_tags: "NormalizedTags") -> "NormalizedTags":
+        return NormalizedTags({**self.tags, **normalized_tags.tags})
+
+
+def validate_tags() -> int: ...
+
+
+def validate_and_normalize_tags(
+    tags: Union[NormalizedTags, Optional[Mapping[str, Any]]],
+    allow_reserved_tags: bool = True,
+    warn_on_deprecated_tags: bool = True,
+    warning_stacklevel: int = 4,
+) -> NormalizedTags:
```

Comment:
Re: naming `normalize` makes me think you are going to mutate the tag keys but that is not what is happening here.

I think we should retain the name `validate_tags` and have the class be `ValidatedTags`

Reply:
The reason I went with normalize is that it actually does change the input, so I think the prior name was misleading. It accepts values that are Any and turns them into strings.

Reply:
Oh awesome. Yes normalize is much better great call.

Just `normalize_tags`?